### ISSUE TITLE
Stakeholder tool: Parent not showing

### ISSUE THREAD

**Issue by sebastiangaertner** [Greenstand/treetracker-admin-client#753]
![Screenshot 2022-07-19 at 15 12 38](https://user-images.githubusercontent.com/3283800/179747546-2b8ebfbb-2388-478c-9763-8492e91e0874.png)

**Comment by Marcelovv2** [comment]
Hi, I'm trying to work on this issue, I have my local environment running, but I can't find the slack channel to get the login. Can someone point me to it? thanks.

**Comment by nmcharlton** [comment]
Hi @Marcelovv2. If you haven't already, you can sign up to contribute to Greenstand using this form: https://docs.google.com/forms/d/e/1FAIpQLSdyLUffLPZZlHR-qfMwch5HkWcZ3kxM6h-eMK5PfGKLD4t8iA/viewform

You'll then get access to Slack. Once you're in, send me a message and I'll add you to the admin_panel_chat channel.

**Comment by iib0011** [comment]
Parents are showing.
![Screenshot (343)](https://user-images.githubusercontent.com/58892076/180878864-3d12331e-83f4-4cd2-895f-ff5ff98fa148.png)

**Comment by Marcelovv2** [comment]
Hi @iib0011, can you tell me how to reproduce this? thanks.

**Comment by iib0011** [comment]
Go to Stakeholders-> Add stakeholder. Fill the form and select a parent. Then search the stakeholder's name you just created

**Comment by Marcelovv2** [comment]
@iib0011 Thanks, I wasn't able to add a stakeholder, I tried local and dev, is this where you added? 
![image](https://user-images.githubusercontent.com/19412353/180884031-6173f348-4507-457e-979c-bdb9dddf59cf.png)

**Comment by iib0011** [comment]
I tried in local

**Comment by nmcharlton** [comment]
This looks like an issue with the Stakeholder API trying to query the (legacy) `public` schema in the DB and not having permission to do so.
@gwynndp @Kpoke Is this a temporary measure for testing and development? If so, we need to make sure the stakeholder DB user has the appropriate permissions.

**Comment by Kpoke** [comment]
When DB users are created for schemas, they are set up to have access to that schema alone, so we would have to give the stakeholder DB user access to the entity table. I'll work on it

**Comment by Kpoke** [comment]
Fixed the issue as regards access to the public schema, at least for the dev environment.

As for the parents not showing, @gwynndp could it be due to the refactor I made on the stakeholder-api? Those parent-children values that I removed?

**Comment by gwynndp** [comment]
Yeah, originally the stakeholder tool would show the direct parents and children for a stakeholder in the detail view and the data was added in arrays for parents and children.  Any children would also have the parent added, and the parent would have the child added so they reflected each other without getting into recursive calls.  

There may be a better way of doing that with one query, or perhaps by adding other endpoints... ideas?

As a sidenote, we probably need to add a /relation endpoint (or something like) to get the related stakeholders for keycloak/auth but I don't know what that will require yet.
